cmd/list.go

```diff
@@ -2,51 +2,110 @@ package cmd
 
 import (
 	"encoding/json"
+	"encoding/xml"
 	"fmt"
-	"gopkg.in/yaml.v2"
 	"io"
-	"os"
+
+	"github.com/ory/viper"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
 
 	"github.com/boson-project/faas"
 	"github.com/boson-project/faas/knative"
-	"github.com/spf13/cobra"
 )
 
-var formats = map[string]fmtFn{
-	"plain": fmtPlain,
-	"json":  fmtJSON,
-	"yaml":  fmtYAML,
+func init() {
+	root.AddCommand(listCmd)
+	listCmd.Flags().StringP("namespace", "n", "", "Override namespace in which to search for Functions.  Default is to use currently active underlying platform setting - $FAAS_NAMESPACE")
+	listCmd.Flags().StringP("output", "o", "plain", "optionally specify output format (plain,json,yaml)")
+
+	err := listCmd.RegisterFlagCompletionFunc("output", CompleteOutputFormatList)
+	if err != nil {
+		fmt.Println("Error while calling RegisterFlagCompletionFunc: ", err)
+	}
+}
+
+var listCmd = &cobra.Command{
+	Use:        "list",
```

Comment:
For consistency, shouldn't this be
```suggestion
	Use:        "list [options]",
```

Reply:
Ah, good catch there too.